File: 10. Subprogram/E. Komposisi Fungsi/e_komposisi_fungsi.java
import java.util.Scanner;
import java.lang.Math;

class e_komposisi_fungsi {
    public static int fx(int A, int B, int x) {
        return Math.abs(A * x + B);
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int A = scan.nextInt();
        int B = scan.nextInt();
        int K = scan.nextInt();
        int x = scan.nextInt();
        scan.close();
        while (K > 0) {
            x = fx(A, B, x);
            K--;
        }
        System.out.println(x);
    }
}

File: 10. Subprogram/E. Komposisi Fungsi/e_komposisi_fungsi.go
package main

import (
  "fmt"
  "math"
)

func fx(A, B, x int) int {
  return int(math.Abs(float64(A*x + B)))
}

func main() {
  var A, B, K, x int
  fmt.Scan(&A, &B, &K, &x)
  for _ = 0; K > 0; K-- {
    x = fx(A, B, x)
  }
  fmt.Println(x)
}
